### DISCUSSION TITLE
RS485

### DISCUSSION THREAD

**Discussion by vc42** [tagyoureit/nodejs-poolController#840] [RS485] [answered]
Hi,

I am an EE in my former life and I am bothered by the way the equipment is connected in the Nixie setup.  The rs-485 bus is supposed to be daisy chained, rather than than starred, and terminated.  How well does it work in the star configuration with let's say three radial connections (a heater, a pump and the chlorinator) ?

**Comment by CodeSlinger69** [comment, marked as answer]
You are correct, a star topology is not recommended for RS485 buses. However, with this incredibly small RS485 network that we are talking about here, I don't think it practically matters. The RS485 topology in my Pentair OCP was a star before I started hacking on it for njsPC. And after hacking on it, it remains a star, as its just easier to bring all the wires together that way.

**Reply by vc42** [reply]
Ok, let me try and see.  I have this long (100') Intelliflo RS-485 cable that I was reluctant to cut, but why not ?  It's not like I will ever have a second pump 100' away.  I heard from someone (that has a multi-pump setup) complaints about pumps occasionally losing communication with his Pentair control system.  When I looked at his connections, I saw a star.  So, Pentair themselves do it even if there are issues with cabling.

**Comment by CodeSlinger69** [comment]
There's a bunch of info [here](https://en.wikipedia.org/wiki/RS-485), but RS485 is specifically used for long distance connections due to its ability to reduce speed signalling if needed. 100 feet is not a lot for RS485, and we don't need high speed signalling for the very little data that is transmitted in this particular pool automation application.

**Reply by vc42** [reply]
Yeah, but I was surprised somewhat by their cavalier approach anyway.  Elk, for example does it properly, so why not pentair ?

Plus, you'd need fewer cables to pull if you used a daisy.  Even irrigation today uses a two-wire system to save on cabling.

**Reply by CodeSlinger69** [reply]
I agree. My irrigation is setup as daisy chain as well. For a Pentair box, we are talking inches of cable runs however. But that said, if we were happy with Pentair, we wouldn't be chatting on this github, eh? Pentair does a lot of silly things, this one doesn't bother me all that much compared to the others.

**Reply by vc42** [reply]
👍
